File: 프로그래머스/1/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {
    var answer = [];
    let p_dict = {}
    
    for(var i=0; i<name.length; i++) {
        p_dict[name[i]] = yearning[i]
    }
    
    for(var i=0; i<photo.length; i++){
        let score = 0
        for(var j=0; j<photo[i].length; j++){
            if(Object.keys(p_dict).includes(photo[i][j])){
                score += p_dict[photo[i][j]]
            } else {
                continue
            }
        }
        answer.push(score)
    }
    
    return answer;
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    answer = []
    # 인물의 그리움 점수를 모두 합산 = 해당 사진의 추억 점수 
    info = list(zip(name,yearning))
    i_dict = {}
    
    for name,score in info:
        i_dict[name] = score
    
    for p_arr in photo:
        score = 0
        for p in p_arr:
            if p in i_dict:
                score += i_dict[p]
            else:
                continue
        answer.append(score)
        
    return answer